SYCL/ESIMD/regression/spirv_Function_Call_Test.cpp

```diff
@@ -0,0 +1,119 @@
+//==------- spirv_Function_Call_Test.cpp - DPC++ ESIMD on-device test-------==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+// REQUIRES: gpu
+// UNSUPPORTED: cuda || hip
+// RUN: %clangxx -fsycl %s -o %t.out
+// RUN: %GPU_RUN_PLACEHOLDER %t.out
+
+// This is a test to validate sycl-post-link correctly lowers calls to spirv
+// builtin globals
+
+#include "../esimd_test_utils.hpp"
+
+#include <sycl/ext/intel/esimd.hpp>
+#include <sycl/sycl.hpp>
+
+#include <cstdint>
+#include <iostream>
+
+using namespace sycl;
+using namespace sycl::ext::intel::esimd;
+
+SYCL_EXTERNAL inline size_t __spirv_GlobalInvocationId_x()
```

Comment:
IMO, if it is E2E test, then it should not use internal functions like these.
They can be replaced with 3 dimensional nd_range passed to kernel and get x,y,z dimensions by nd_item.get_global_id(dimension)

Otherwise, if it is not E2E test, it would be better to create a test accepting LLVM IR